.github/workflows/unit_tests.yml

```diff
@@ -4,6 +4,8 @@ name: Run Unit Tests
 on:
   pull_request:
   push:
```

Comment:
Do we want to run on push at all if we run on PR?

Reply:
You would know better than I, but I thought we had to do this to make it run directly on dev after a merge. Outside metrics and tools that reference the repo's CI status should probably reference the dev branch, so something needs to trigger it.

Ex: if something other than Actions does CD, or the README badges for that matter

Reply:
Ah, that is how to make that happen, though on `pull_request` will test any commits to a PR which theoretically handles all of those except a commit directly to `dev`, which should probably be forbidden in branch protection rules?

Maybe unit tests on PR and coverage on push? This would prevent checking coverage on every commit and prevent duplicating tests when you open PR and then merge the same?

Reply:
Oh, it should now be running only on "push" (merge) to `dev`:

```yaml
name: Run Unit Tests
on:
  pull_request:
  push:
    branches:
      - dev
```

So only once per push for PR branches. If I've set it up the way I think, this will run every time you breathe on a PR, and whenever you merge a PR into dev, but it won't fire on non-PR branches.

I dunno about the coverage thing. I agree that every time you `git push` could be excessive, but I wanna see it before a merge.

Reply:
That is correct, just means that opening a PR and merging that PR will call the same workflow which is definitely excessive, though we probably aren't too concerned about hitting action limits right now..

Note that a 'push' could be a force push or a commit; I don't think this should happen on the dev branch, but not sure if those branch protection rules are there. Also note that a draft PR is a PR, so if you always open a PR when you start a branch (I do), you will run actions on every commit (good for unit tests IMO, not sure for coverage).

Reply:
Is there a way to reference status on `dev` without running the suite on `dev`? It's mainly the unit and build runs whose status I'd like to monitor. Bragging about coverage is just a nice plus 😉

Reply:
Hmm.. well, GH Actions are tied to commit ID's, so the run needs to be on whatever the latest commit on `dev` is..